src/components/cc-domain-management/cc-domain-management.types.d.ts

```diff
@@ -0,0 +1,89 @@
+export type DomainManagementFormState = DomainManagementFormStateIdle | DomainManagementFormStateAdding;
+
+interface DomainManagementFormStateIdle {
+  type: 'idle';
+  hostname: {
+    value: string;
+    error?: FormError | null;
+  };
+  pathPrefix: {
+    value: string;
+  };
+}
+
+interface DomainManagementFormStateAdding {
+  type: 'adding';
+  hostname: {
+    value: string;
+    error?: null;
+  };
+  pathPrefix: {
+    value: string;
+  };
+}
+
+export type FormError = 'empty' | 'invalid-format' | 'invalid-wildcard' | HostnameContainsPathError;
```

Comment:
question: is it a problem to have both string or object?

Reply:
I don't have a definite answer for that, I felt it was weird but it does create issues.

The alternative would be to go for objects for all of these but it also feels weird to go for objects like this:
```js
{ code: "empty" }
```

This solution however makes dealing with FormErrors a little bit simpler because we know they always have the same shape...

If you're fine with that I think I'd go with that (all errors are objects with at least a `code` property).

Reply:
Go for "all errors are objects" :muscle:

Reply:
+1, it would make it clearer